compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpCompilerPluginUtil.java

```diff
@@ -209,10 +207,16 @@ public static TypeDescKind retrieveEffectiveTypeDesc(TypeSymbol descriptor) {
         return typeDescKind;
     }
 
-    private static boolean isBasicTypeDesc(TypeDescKind kind) {
-        return kind == TypeDescKind.STRING || kind == TypeDescKind.INT || kind == TypeDescKind.FLOAT ||
-                kind == TypeDescKind.DECIMAL || kind == TypeDescKind.BOOLEAN || kind == TypeDescKind.JSON ||
-                kind == TypeDescKind.XML || kind == TypeDescKind.RECORD || kind == TypeDescKind.BYTE;
+    public static boolean isAnyDataType(TypeDescKind kind) {
+        return kind == TypeDescKind.BOOLEAN || kind == TypeDescKind.INT || kind == TypeDescKind.FLOAT ||
```

Comment:
Rather than having a huge boolean condition can't we have a list of all the types included in `anydata` and verify whether the given type is available in that list ? I feel that it would be much more readable than this approach.